docs/bokeh/source/docs/user_guide/topics/graph.rst

```diff
@@ -39,8 +39,8 @@ The following code snippet
 .. code-block:: python
 
     import math
-    from bokeh.plotting import figure
-    from bokeh.models import GraphRenderer, Ellipse
+    from bokeh.plotting import figure, show, output_file
+    from bokeh.models import GraphRenderer, Ellipse, StaticLayoutProvider
```

Comment:
@mosc9575 I don't think `StaticLayoutProvider` is used in this snippet (or `show` or `output_file` for that matter, we have definitely been omitting the optional `output_file` in most examples unless we explicitly mean to demonstrate it in particular)

Reply:
This is an example which is splitted into two code blocks. The second code block begins in line 77. The complete example is

```python
import math
from bokeh.plotting import figure
from bokeh.plotting import figure, show, output_file
from bokeh.models import GraphRenderer, Ellipse, StaticLayoutProvider
from bokeh.palettes import Spectral8

# list the nodes and initialize a plot
N = 8
node_indices = list(range(N))
plot = figure(title="Graph layout demonstration", x_range=(-1.1,1.1),
                      y_range=(-1.1,1.1), tools="", toolbar_location=None)
graph = GraphRenderer()
# replace the node glyph with an ellipse
# set its height, width, and fill_color
graph.node_renderer.glyph = Ellipse(height=0.1, width=0.2,
                                        fill_color="fill_color")
# assign a palette to ``fill_color`` and add it to the data source
graph.node_renderer.data_source.data = dict(
        index=node_indices,
        fill_color=Spectral8)

# generate ellipses based on the ``node_indices`` list
circ = [i*2*math.pi/8 for i in node_indices]
# create lists of x- and y-coordinates
x = [math.cos(i) for i in circ]
y = [math.sin(i) for i in circ]
# convert the ``x`` and ``y`` lists into a dictionary of 2D-coordinates
# and assign each entry to a node on the ``node_indices`` list
graph_layout = dict(zip(node_indices, zip(x, y)))
# use the provider model to supply coourdinates to the graph
graph.layout_provider = StaticLayoutProvider(graph_layout=graph_layout)
# render the graph
plot.renderers.append(graph)
# specify the name of the output file
output_file('graph.html')
# display the plot
show(plot)
```

To run the example, the imports I added were missing. I don't have a problem to update the example and/or remove the `output_file`. Please tell me what you prefer.

Reply:
Oh I see. This is probably a good indication why "split" examples might be discouraged. In that case, I'd suggest just removing the `output_file` as it is not strictly needed, and this example is already on the long side.